rollup/missing_header_fields/export-headers-toolkit/README.md

```diff
@@ -0,0 +1,64 @@
+# Export missing block header fields toolkit
+
+A toolkit for exporting and transforming missing block header fields of Scroll before {{upgrade_name}} TODO: replace when upgrade is clear.
+
+## Context
+We are using the [Clique consensus](https://eips.ethereum.org/EIPS/eip-225) in Scroll L2. Amongst others, it requires the following header fields:
+- `extraData`
+- `difficulty`
+
+However, before {{upgrade_name}}, these fields were not stored on L1/DA.
+In order for nodes to be able to reconstruct the correct block hashes when only reading data from L1, 
+we need to provide the historical values of these fields to these nodes through a separate file.
+
+This toolkit provides commands to export the missing fields, deduplicate the data and create a file 
+with the missing fields that can be used to reconstruct the correct block hashes when only reading data from L1.
+
+The toolkit provides the following commands:
+- `fetch` - Fetch missing block header fields from a running Scroll L2 node and store in a file
+- `dedup` - Deduplicate the headers file, print unique values and create a new file with the deduplicated headers 
+
+## Binary layout deduplicated missing header fields file
+The deduplicated header file binary layout is as follows:
+
+```plaintext
+<unique_vanity_count:uint8><unique_vanity_1:[32]byte>...<unique_vanity_n:[32]byte><header_1:header>...<header_n:header>
+
+Where:
+- unique_vanity_count: number of unique vanities n
+- unique_vanity_i: unique vanity i
+- header_i: block header i
+- header: 
+    <vanity_index:uint8><flags:uint8><seal:[65|85]byte>
+    - vanity_index: index of the vanity in the unique vanity list
+    - flags: bitmask, lsb first
+        - bit 0: 0 if difficulty is 2, 1 if difficulty is 1
+        - bit 1: 0 if seal length is 65, 1 if seal length is 85
+        - rest: 0
```

Comment:
what if store `flags` and `vanity_index` in 1 byte instead od 2? This will not help to save a lot, for current  mainnet it's ≈ 7MB, just for thinking. 

Also, maybe it worth to completely remove that `vanity` from `ExtraData` field, because it's fully unused and random. In that case this `vanity` byte will become useless too

Reply:
Good point, initially wanted to add all of the information to the bitmask but then somehow got sidetracked by the amount of vanities and didn't check whether it fits or not. I added it in https://github.com/scroll-tech/go-ethereum/pull/903/commits/d7af40a9d7d884358bd9018c785ebaabfd357832